### DISCUSSION TITLE
React crashes with 	Warning: Unexpected ref object provided for input. Use either a ref-setter function or React.createRef().

### DISCUSSION THREAD

**Discussion by joe-oli** [react-hook-form/react-hook-form#8253] [Q&A] [answered]
When using  ` <input ref={register()}  //etc />` , my app is crashing, and console shows the error as per the above title.

This usage of ref appears perfectly valid, as per the the example video at 
https://www.youtube.com/watch?v=LhIvi4XHIco
and equivalent sandbox at
https://codesandbox.io/s/cocky-night-mqe9g

The difference is I am using RHF v7 but the tutorial is for RHF v6; Is that the reason for the crash? ( and if so, how to do the equivalent with RHF v7 )

**Comment by Moshyfawn** [comment, marked as answer]
In V7, the `register` has changes.

Now you need to spread the register props into your input

```jsx
<input {...register("fieldName")} />
```

Please, check out [the RHF docs](https://react-hook-form.com/api/useform/register) to learn more about the new API
